### PR TITLE
feat!: migrate to GitHub

### PR DESCRIPTION
**Pull request by sanakhandvsa**
## Description

Related issue: [JIRA_TICKET_NUMBER](LINK_TO_JIRA_TICKET)

## Before submitting (or marking as "ready for review")

- [ ] Does the pull request title follow the [conventional commit](https://www.conventionalcommits.org/en/v1.0.0/) specification?
- [ ] Have you performed a self-review of the code
- [ ] Have you have added tests that prove the fix or feature is effective and working
- [ ] Did you make sure to update any documentation relating to this change?

### CODE REVIEW

File: test/Common/src/Common/Form/Elements/Custom/DateTimeSelectTest.php [hunk 1]
```diff
@@ -47,7 +47,7 @@ public function testSetValueStringUtc()
         $this->assertSame('2016', $this->sut->getYearElement()->getValue());
         $this->assertSame('06', $this->sut->getMonthElement()->getValue());
         $this->assertSame('14', $this->sut->getDayElement()->getValue());
-        $this->assertSame('15', $this->sut->getHourElement()->getValue());
+        $this->assertSame(15, $this->sut->getHourElement()->getValue()+1);
```
Comment (fibble) [inline test/Common/src/Common/Form/Elements/Custom/DateTimeSelectTest.php:50]: not an issue, just wondering why this change is needed?

**Review by fibble — APPROVED** [review]

File: composer.json [hunk 2]
```diff
@@ -2,40 +2,46 @@
...
+        "laminas/laminas-i18n": "^2.14",
+        "laminas/laminas-modulemanager": "^2.9",
+        "laminas/laminas-mvc": "^2.7",
+        "laminas/laminas-view": "^2.11",
+        "laminas/laminas-validator": "^2.11",
+        "olcs/olcs-utils": "^5.0",
+        "dvsa/authentication-cognito": "^2.0",
+        "laminas/laminas-console": "^2.7",
+        "laminas/laminas-stdlib": "^2.7",
+        "laminas/laminas-zendframework-bridge": "^1.6",
```
Comment (JoshuaLicense) [inline composer.json:22]: Does this library depend on this?

File: composer.json [hunk 3]
```diff
@@ -2,40 +2,46 @@
...
+        "laminas/laminas-eventmanager": "^2.6",
+        "laminas/laminas-http": "^2.8",
+        "laminas/laminas-i18n": "^2.14",
+        "laminas/laminas-modulemanager": "^2.9",
+        "laminas/laminas-mvc": "^2.7",
+        "laminas/laminas-view": "^2.11",
+        "laminas/laminas-validator": "^2.11",
+        "olcs/olcs-utils": "^5.0",
+        "dvsa/authentication-cognito": "^2.0",
+        "laminas/laminas-console": "^2.7",
```
Comment (JoshuaLicense) [inline composer.json:20]: Does this library depend on this?

File: composer.json
Comment (JoshuaLicense) [inline composer.json:1]: Are we missing Laminas filter from the dependencies?

File: composer.json [hunk 4]
```diff
@@ -2,40 +2,46 @@
...
+        "laminas/laminas-stdlib": "^2.7",
+        "laminas/laminas-zendframework-bridge": "^1.6",
+        "container-interop/container-interop": "^1.2",
+        "laminas/laminas-psr7bridge": "^0.2.2"
```
Comment (JoshuaLicense) [inline composer.json:24]: Is this required?

File: composer.json [hunk 5]
```diff
@@ -2,40 +2,46 @@
...
     },
     "require-dev": {
+        "bamarni/composer-bin-plugin": "^1.8",
+        "dms/phpunit-arraysubset-asserts": "^0.5.0",
         "doctrine/annotations": "^1.14.2",
-        "olcs/olcs-devtools": "~4.206.0",
+        "ezyang/htmlpurifier": "^4.17",
```
Comment (JoshuaLicense) [inline composer.json:30]: This seems to be used in source code rather than just tests

File: composer.json [hunk 6]
```diff
@@ -2,40 +2,46 @@
...
+        "johnkary/phpunit-speedtrap": "^4.0",
+        "laminas/laminas-code": "^3.4.1",
```
Comment (JoshuaLicense) [inline composer.json:32]: Used in source code?

File: composer.json [hunk 7]
```diff
@@ -2,40 +2,46 @@
...
+        "laminas/laminas-navigation": "2.13.0",
```
Comment (JoshuaLicense) [inline composer.json:33]: Used in source code.

File: composer.json [hunk 8]
```diff
@@ -2,40 +2,46 @@
...
+        "laminas/laminas-serializer": "^2.5",
```
Comment (JoshuaLicense) [inline composer.json:34]: Is this used at all?

File: composer.json [hunk 9]
```diff
@@ -2,40 +2,46 @@
...
-        "olcs/olcs-transfer": "~4.206.0",
-        "olcs/olcs-utils": "^5.0.0",
-        "phpstan/phpstan": "^0.12.100",
+        "olcs/olcs-transfer": "^5.0.0",
+        "phpunit/phpunit": "^9.6",
         "soflomo/purifier": "^1.0",
-        "symfony/lock": "^5"
+        "symfony/filesystem": "^5.4",
+        "symfony/lock": "^5",
+        "laminas/laminas-inputfilter": "^2.10"
```
Comment (JoshuaLicense) [inline composer.json:44]: Used in source code.

**Review by JoshuaLicense — CHANGES_REQUESTED** [review]

File: test/Common/src/Common/Form/Elements/Custom/DateTimeSelectTest.php (hunk 1 above)
Comment (sanakhandvsa) [reply test/Common/src/Common/Form/Elements/Custom/DateTimeSelectTest.php:50]: Assertion was failing because UTC is always 1 hr ahead to Europe/London timezone.
we might need to do permanent fix later to accommodate  both when time changes .

File: test/phpunit.xml
[the remark this answers is not in the document]
Comment (sanakhandvsa) [reply test/phpunit.xml:1]: I have tested not needed,

File: test/Common/src/Common/Form/Elements/Custom/DateTimeSelectTest.php (hunk 1 above)
Comment (JoshuaLicense) [reply test/Common/src/Common/Form/Elements/Custom/DateTimeSelectTest.php:50]: This will fail for 6 months of the year though right? Whereas before it didn't.

We need to either align the code to work in UTC or set the tests back to the Europe/London zone.

File: composer.json (hunk 2 above)
Comment (JoshuaLicense) [reply composer.json:22]: Can we do a pass over the dependencies please

File: composer.json (hunk 3 above)
Comment (sanakhandvsa) [reply composer.json:20]: yes validator and utils been used in source code

Comment (JoshuaLicense) [reply composer.json:20]: Sorry Sana, these comments only relate the the line they are attached to, so in this case, line 20.

File: composer.json
Comment (sanakhandvsa) [reply composer.json:1]: Ah great catch added

File: composer.json (hunk 4 above)
Comment (sanakhandvsa) [reply composer.json:24]: "laminas/laminas-stdlib": "^2.7", been used in SystemInfoMessagesTest
 "container-interop/container-interop": "^1.2", is been used all places at ServiceFactory 
 psr7bridge:  removed

File: composer.json (hunk 5 above)
Comment (sanakhandvsa) [reply composer.json:30]: yes moved to source code

File: composer.json (hunk 8 above)
Comment (sanakhandvsa) [reply composer.json:34]: Laminas\Serializer\AdapterPluginManager has been used in AbstractFormValidationTestCase:63
 $this->sut = $this->getForm()

File: composer.json (hunk 9 above)
Comment (sanakhandvsa) [reply composer.json:44]: filesystem, symfony lock  has been used in test files

File: composer.json (hunk 3 above)
Comment (sanakhandvsa) [reply composer.json]: ya console been removed ,

File: composer.json (hunk 6 above)
Comment (sanakhandvsa) [reply composer.json]: yes

File: composer.json (hunk 9 above)
Comment (JoshuaLicense) [reply composer.json]: `laminas/laminas-inputfilter` is used in the source code. Could we move it to `dependencies` please?

File: composer.json (hunk 8 above)
Comment (JoshuaLicense) [reply composer.json]: Sorry, I'm missing something, what do you mean?

How is: `Laminas\Serializer\AdapterPluginManager` related to: `$this->sut = $this->getForm()`

Comment (sanakhandvsa) [reply composer.json]: i mean `Laminas\Serializer` moved to dev dependencies

File: composer.json (hunk 9 above)
Comment (sanakhandvsa) [reply composer.json]: moved

File: composer.json (hunk 8 above)
Comment (JoshuaLicense) [reply composer.json]: I'm still lost.

Comment (sanakhandvsa) [reply composer.json]: Looks like `laminas/laminas-mvc` dependent and requires  `Laminas\Serializer`
Tests are failing 
Screenshot 2023-12-13 at 12 22 31

Comment (sanakhandvsa) [reply composer.json]: `Error: Class 'Laminas\Serializer\AdapterPluginManager' not found

/Users/sanaullahkhan/Documents/code/olcs/olcs-common/vendor/laminas/laminas-mvc/src/Service/AbstractPluginManagerFactory.php:35
/Users/sanaullahkhan/Documents/code/olcs/olcs-common/vendor/laminas/laminas-mvc/src/Service/AbstractPluginManagerFactory.php:48
/Users/sanaullahkhan/Documents/code/olcs/olcs-common/vendor/laminas/laminas-servicemanager/src/ServiceManager.php:933
/Users/sanaullahkhan/Documents/code/olcs/olcs-common/vendor/laminas/laminas-servicemanager/src/ServiceManager.php:1091
/Users/sanaullahkhan/Documents/code/olcs/olcs-common/vendor/laminas/laminas-servicemanager/src/ServiceManager.php:633
/Users/sanaullahkhan/Documents/code/olcs/olcs-common/vendor/laminas/laminas-servicemanager/src/ServiceManager.php:596
/Users/sanaullahkhan/Documents/code/olcs/olcs-common/vendor/laminas/laminas-servicemanager/src/ServiceManager.php:529
/Users/sanaullahkhan/Documents/code/olcs/olcs-common/vendor/laminas/laminas-modulemanager/src/Listener/ServiceListener.php:206
/Users/sanaullahkhan/Documents/code/olcs/olcs-common/vendor/laminas/laminas-eventmanager/src/EventManager.php:489
/Users/sanaullahkhan/Documents/code/olcs/olcs-common/vendor/laminas/laminas-eventmanager/src/EventManager.php:247
/Users/sanaullahkhan/Documents/code/olcs/olcs-common/vendor/laminas/laminas-modulemanager/src/ModuleManager.php:128
/Users/sanaullahkhan/Documents/code/olcs/olcs-common/test/Common/src/Common/Controller/Lva/AbstractFormValidationTestCase.php:1304
/Users/sanaullahkhan/Documents/code/olcs/olcs-common/test/Common/src/Common/Controller/Lva/AbstractFormValidationTestCase.php:75
/Users/sanaullahkhan/Documents/code/olcs/olcs-common/test/Common/src/Common/Controller/Lva/AbstractFormValidationTestCase.php:140
/Users/sanaullahkhan/Documents/code/olcs/olcs-common/test/Common/src/Common/Controller/Lva/AbstractFormValidationTestCase.php:63
`

Comment (JoshuaLicense) [reply composer.json]: Okay, so this is a Laminas MVC issue as it's is missing a dependency? Fair enough. Keep it in `devDependencies` but please note when we're doing the v3 upgrade to remove it if it's fixed upstream.

File: composer.json (hunk 3 above)
Comment (JoshuaLicense) [reply composer.json]: Why was `dvsa/authentication-cognito` removed?

File: test/Common/src/Common/Form/Elements/Custom/DateTimeSelectTest.php [hunk 10]
```diff
@@ -77,7 +77,7 @@ public function testSetValueDateTimeUtc()
...
```
Comment (JoshuaLicense) [inline test/Common/src/Common/Form/Elements/Custom/DateTimeSelectTest.php:80]: Could this be `'14'` too?

**Review by JoshuaLicense — CHANGES_REQUESTED** [review]
Looks good! Just a small question added re: consistency in the test.

**Review by fibble — DISMISSED** [review]

**Review by JoshuaLicense — APPROVED** [review]
